### PR TITLE
Changes

### PR DESCRIPTION
**Pull request by bretsky**
- Gitmodule stuff fixed
 - Fix `solenoid.cpp`
 - Camera display now stays inside the window
 - Finer zoom

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,6 @@
 [submodule "include/googletest"]
```
Comment (Mogball) [inline .gitmodules]: OHH this is why it's my fault

**Comment by Mogball** [comment]
gonna make sure I test my builds until this travis thing gets working

**Comment by bretsky** [comment]
@mogball did you not try building it XD?

**Comment by Mogball** [comment]
I think that's when my PC died and I just merged it on my phone loool

**Comment by Mogball** [comment]
and I do __NOT__ appreciate the uppercase xd

**Comment by bretsky** [comment]
Is eCkSdEe a suitable compromise?

**Comment by Mogball** [comment]
yes

**Pull request merged**